string name
bool[] seen
int64 image_x
int64 image_y
float64 x
float64 y
float64 z
bool added
